src/rgw/rgw_rados.h

```diff
@@ -3567,7 +3567,7 @@ class RGWRados
   }
 
   bool need_to_log_metadata() {
-    return is_meta_master() && get_zone().log_meta;
+    return is_meta_master() && get_zone().log_meta && get_zonegroup().zones.size() > 1;
```

Comment:
we still want this to return true in a configuration that has two zonegroups, each with one zone. the secondary zonegroup would still sync metadata from the meta master